src/components/learning-log.tsx

```diff
@@ -1,16 +1,46 @@
-import { observer } from "mobx-react";
+import { inject, observer } from "mobx-react";
 import * as React from "react";
 
+import { IAllStores } from "..";
+import { UIModelType } from "../models/ui";
 import "./learning-log.sass";
+import { TabComponent } from "./tab";
+import { TabSetComponent } from "./tab-set";
 
+interface IInjectedProps {
+  ui: UIModelType;
+}
+
+@inject((allStores: IAllStores) => {
+  const injected: IInjectedProps = {
+    ui: allStores.ui,
+  };
+  return injected;
+})
 @observer
 export class LearningLogComponent extends React.Component<{}, {}> {
 
+  get injected() {
+    return this.props as IInjectedProps;
```

Comment:
No, this is just to make TypeScript happy - you can cast to any type, even ones that don't share any attributes.

Reply:
Got it - so is the idea that `.injected` will technically include *all* `props`, but TypeScript will only let us access the injected ones because of this cast?

Reply:
You got it.